pkg/analyzer/gateway.go

```diff
@@ -41,7 +43,15 @@ func (GatewayAnalyzer) Analyze(a common.Analyzer) ([]common.Result, error) {
 	if err != nil {
 		return nil, err
 	}
-	if err := client.List(a.Context, gtwList, &ctrl.ListOptions{}); err != nil {
+	labelSelectorMap := make(map[string]string)
```

Comment:
Probably worth creating a util function since you replicate it for 3 analysers and is likely more CR analysers from the new gateway k8s api will be introduced  e.g TCPRoute

Reply:
good! I added on some analyzer